lib/puma/client.rb

```diff
@@ -183,12 +186,20 @@ def decode_chunk(chunk)
         if line.end_with?("\r\n")
           len = line.strip.to_i(16)
           if len == 0
+            @in_last_chunk = true
             @body.rewind
             rest = io.read
-            rest = rest[2..-1] if rest.start_with?("\r\n")
-            @buffer = rest.empty? ? nil : rest
-            set_ready
-            return true
+            last_crlf_size = "\r\n".bytesize
+            if rest.bytesize < last_crlf_size
```

Comment:
this feels pretty brittle too. aren't there edge cases where this would be `true`?

Reply:
There are edge cases. For example, the first request omits the last empty chunk:

```ruby
sock = TCPSocket.new @host, @server.connected_port
# No last empty chunk
sock << "GET / HTTP/1.1\r\nConnection: Keep-Alive\r\nTransfer-Encoding: chunked\r\n\r\n1\r\nh\r\n4\r\nello\r\n0\r\n"
# The leading "GE" is consumed as the last empty chunk (\r\n)
sock << "GET / HTTP/1.1\r\nConnection: Keep-Alive\r\nTransfer-Encoding: chunked\r\n\r\n4\r\ngood\r\n3\r\nbye\r\n0\r\n\r\n"

h = header(sock)
assert_equal ["HTTP/1.1 200 OK", "Content-Length: 0"], h
assert_equal "hello", body

h = header(sock)
# The next request starts with "T / HTTP/..." (No "GE")
assert_equal ["HTTP/1.1 200 OK", "Content-Length: 0"], h
assert_equal "goodbye", body

sock.close
```

I omitted robust content check to follow the current style.

I think that robust content check should be done in a separated pull request.

Reply:
OK